core/src/main/java/de/bwaldvogel/mongo/backend/MongoBackendClock.java

```diff
@@ -0,0 +1,60 @@
+package de.bwaldvogel.mongo.backend;
+
+import de.bwaldvogel.mongo.bson.BsonTimestamp;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class MongoBackendClock extends Clock {
```

Comment:
What do you need a separate class for `Clock` ?

Reply:
For `BsonTimestamp` probably you need to create an interface like below:

```java
interface Sequencer {
   /** Generate next operation timestamp according to mongo rules **/
   BsonTimestamp next();
}
```

There can be several implementations for the sequencer. One of which  will allow you to  inject / modify `Clock` and sequence number for tests

Reply:
It is same rule again: prefer composition over inheritance.

Reply:
In 70387efbaa118a8e49ecf6fa27f5b882d94aa4a0 I’ve converted the `MongoBackendClock` into a simple (non-public) `OplogClock` that just **has-a** clock .